sktime/forecasting/var_reduce.py

```diff
@@ -0,0 +1,334 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements ReducedVAR where user can choose the type of Regressor to use."""
+
+# __author__ = [meraldoantonio]
+
+import numpy as np
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class VARReduce(BaseForecaster):
+    """
+    VAR-like reduced forecaster.
+
+    It reduces multivariate time series to tabular regression and trains a regressor.
+    Any scikit-learn compatible regressor can be used; by default, `LinearRegression`,
+    is used, making it behave like a traditional VAR. Users can specify other
+    regressors such as `ElasticNet` to introduce regularization and potentially
+    enhance performance with large datasets or in the presence of multicollinearity.
+
+    Parameters
+    ----------
+    lags : int, optional, default=1
+        The number of lagged values to include in the model.
+    regressor : object, optional (default=LinearRegression())
+        The regressor to use for fitting the model. Must be scikit-learn-compatible.
+
+    Attributes
+    ----------
+    coefficients : np.ndarray, shape (lags, num_series, num_series)
+        The estimated coefficients of the model
+    intercept : np.ndarray, shape (num_series,)
+        The intercept term of the model for each time series.
+    num_series : int
+        The number of time series being modeled.
+
+    Examples
+    --------
+    >>> from sktime.forecasting.var_reduce import VARReduce
+    >>> from sktime.datasets import load_longley
+    >>> _, y = load_longley()
+    >>> forecaster = VARReduce()  # doctest: +SKIP
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    VARReduce(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])  # doctest: +SKIP
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["meraldoantonio"],
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+    }
+
+    def __init__(self, lags=1, regressor=None):
+        from sklearn.base import clone
+        from sklearn.linear_model import LinearRegression
+
+        self.regressor = regressor
+        self.lags = lags
+        if regressor is None:
+            self.regressor_ = LinearRegression()
+        else:
+            self.regressor_ = clone(regressor)
+
+        assert hasattr(self.regressor_, "fit"), "Regressor must have 'fit'"
+        assert hasattr(self.regressor_, "predict"), "Regressor must have 'predict'"
+
+        # a dictionary of var_name: fitted regressor, to be filled in during fitting
+        self.regressors = {}
+        self.coefficients_ = None
+        self.intercept_ = None
+        self.num_series = None
+        super().__init__()
+
+    def prepare_for_fit(self, data, return_as_ndarray=True):
```

Comment:
can you kindly make this private? Just add an underscore to the name.
(because we would not expect users to call this)

Reply:
(Resolved)